Code/IO/src/sitkImageFileReader.cxx

```diff
@@ -66,21 +80,154 @@ namespace itk {
       return this->m_FileName;
     }
 
-    Image ImageFileReader::Execute () {
+
+    void
+    ImageFileReader
+    ::UpdateImageInformationFromImageIO(const itk::ImageIOBase* iobase)
+    {
+      PixelIDValueType pixelType;
+      this->GetPixelIDFromImageIO(iobase, pixelType, m_Dimension);
+
+      std::vector<double> direction;
+
+      direction.reserve(m_Dimension*m_Dimension);
+
+      std::vector<double> origin(m_Dimension);
+      std::vector<double> spacing(m_Dimension);
+      std::vector<uint64_t> size(m_Dimension);
+      for( unsigned int i = 0; i < m_Dimension; ++i)
+        {
+        origin[i] = iobase->GetOrigin(i);
+        spacing[i] = iobase->GetSpacing(i);
+        size[i] = iobase->GetDimensions(i);
+        const  std::vector< double > temp_direction = iobase->GetDirection(i);
+        direction.insert(direction.end(), temp_direction.begin(), temp_direction.end());
+        }
+
+      // release functions bound to old meta data dictionary
+      if (m_MetaDataDictionary.get())
+        {
+        this->m_pfGetMetaDataKeys = SITK_NULLPTR;
+        this->m_pfHasMetaDataKey = SITK_NULLPTR;
+        this->m_pfGetMetaData =  SITK_NULLPTR;
+        }
+
+      this->m_MetaDataDictionary.reset(new MetaDataDictionary(iobase->GetMetaDataDictionary()));
+
+      m_PixelType = static_cast<PixelIDValueEnum>(pixelType);
+
+      // Should this be as reported by ImageIO, or as reported by sitk::Image?
```

Comment:
I would go with the number reported by ImageIO. I assume here this number is related to the file format (RGB==3). This is consistent with the code returning information with respect to file types that SimpleITK doesn't support loading.